### PR TITLE
Civilian+ Ballistic Armor Additions

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Content "Civilian+ Ballistic Armor Additions"

#### Purpose of change
This is part of my bugout bag additions. Goal is to add more modern armor and diversify the options that are available while keeping things as simple as possible. Lots of customization without much additional complexity. Lots of stuff worse (but lighter) than the US Ballistic Vest that civilians could have their hands on reasonably.

#### Describe the solution
Infrastructure:
I've Added Some C++ Infrastructure so that pockets added to Load Bearing Vests and Ballistic Vests at spawn attach as pockets. Meaning each vest spawns unique.

I've added some infrastructure for appending variant descriptions for items instead of having to recopy all the same bits. This should make it easier to translate large numbers of variants in the future.

Content Additions:

- [x] Soft armor stab vest
- [x] level 3a armor vest
- [x] level 3 armor vest
- [x] level 3a soft plates
- [x] level 3a hard steel plates
- [x] Riot ballistic vest
- [x] 3 Light Ballistic Vest versions with separate pocket variants
- [x] Balance to number of pockets each LBV can hold (they are all a little low right now)
- [x] Spawns for the above
- [x] ~Police armor rework to be more integrated into other load outs and make sense with the new modern armor additions.~ scoped out partially
- [x] ~Riot Ballistic Vest~ This is also getting scoped out
- [x] SWAT Ballistic Vest
- [x] ~MOLLE spawn groups with a pouch with themed stuff in them. Example a dump pouch with glow sticks, h2o pouch with water bottle etc.~ gonna boot this out of scope
- [x] Variants! Different colors / descriptions for Light and Medium vest (the riot ballistic vest will be a US Ballistic Vest variant)
- [x] ~Named LBV's with default specific pouches in their spawn group. So the "Zombie Hunter" will be a light LBV with a shotshell pouch and a gas mask pouch *always*~ gonna move this to another PR
- [x] Some LBV overhauls included in spawn lists
- [x] Spawns where relevant for the above

Details
<details>
  <summary>Stab Vest: Comes in two variants "security stab vest" and "airsoft vest"</summary>
  
image
image
</details>

<details>
  <summary>level 3a armor vest AKA "soft armor vest": Comes in two variants "security ballistic vest" and "press vest"</summary>
  
image
image
</details>

<details>
  <summary>level 3 armor vest AKA "hard armor vest": Comes in three variants "humanitarian ballistic vest", "police ballistic vest", "national guard ballistic vest" </summary>
  
image
image
image

</details>

<details>
  <summary>Soft stab plate: not very structurally tough but light and unencumbering.</summary>
  
image
</details>

<details>
  <summary>LBV Balance:</summary>
  
After hours staring at LBV's and ballistic vests to figure out what people put on them, the numbers I had were just too low. There were obvious balance odities like LBVs that only take up the upper torso having more storage the ballistic vests on the whole torso. Values are higher across the board. When belts get delayered the heavy LBV will also count as a belt.

Light Armor - 6 (lots of molle to work with all upper torso)
Medium / Heavy Armor - 10 lots of molle to work with including on your sides and abdomen
Light LBV - 4 slots, people can run about 4 mag pouches abreast on these really light rigs
Normal LBV - 8 slots, only upper torso buuuuut it wraps around in a way that most light armor doesn't giving some more space
Heavy LBV - 14 slots, takes up lower torso and also will in the future count as wearing a belt. 
</details>

<details>
  <summary>Steel plates in 3a and 3:</summary>

  image
image

</details>

<details>
  <summary>New Light Ballistic Vests</summary>

So 3 overall light ballistic vests: 
the current one with a cummerbund for magazines
image

a new one with similar sized pouch but with a zipper for normal stuff
image

a new one with instead a shoulder pocket for a radio and a hook for a glow stick
image

Since Light Ballistic Vests AKA plate carriers are very popular and see a lot of ownership the above come in a number of varieties (already seen partially above):

at a weight of 4 are: Olive, Tan, and Black/Grey
at a smaller weight of 1 are themed vests: Future (hexagonal), Punisher (skull) and Meme (gaudy)

Examples of the blurbs are as follows:
image
image
image
image
image
image

</details>

<details>
  <summary>SWAT Armor</summary>

In the US the SWAT and elite police are decked out like a paramilitary force. As such the SWAT armor has been deprecated and replaced as a variant of the US Ballistic Vest. Spawn groups have been added to make sure Soldiers don't spawn with SWAT armor and Police don't spawn with Military armor.

image

</details>

<details>
  <summary>Spawns</summary>

Added spawn pool for all armor inserts called "armor_plates", military surplus stores can have these on shelves.

Stab Vests: Airsoft in sports stores, any in survivor armor (on farmhouses)

Soft 3A vests: Lab security has a new spawn pool with this or a light ballistic vest, military surplus for the security version. 

New Distribution for unisex coats that adds the chance for someone to own a stab vest or soft 3A vest. That also means they might have had em on during zday.

Level 3 armor: Cops now have split groups for armor and other gear, For armor they can spawn with Riot armor, Kevlar concealed vests, or more than likely a police ballistic vest (the new item). 

No humanitarian spawns at this point

Soldiers are likely to spawn with shittier national guard vests instead of full military ones.

</details>

<details>
  <summary>Pouch Changes</summary>

I've made it so that on spawn, molle compatible items spawn with an amount of pouches already attached. Its guaranteed to be at least one ammo pouch per size and then a few others. Should make the world feel more lived in and sensible. Other changes to LBV's are getting pushed back to another PR this one is already way too big.

*this also effects MOLLE packs as well and I've reorganized their spawns to work for it*

I've consolidated LBV spawns as well so the game either spawns a soldier with a naked ballistic vest and a loaded LBV or just a loaded ballistic vest. You never get 2 vests covered in pouches.

got this right off a zombie 
image

</details>

#### Describe alternatives you've considered
Traveling back in time.

#### Testing
Game runs and drops all the stuff it should where it should.

### CODE REVIEW

**Comment by TheShadowFerret** [comment]
>Currently I've Added Some C++ Infrastructure so that pockets added to Load Bearing Vests and Ballistic Vests at spawn attach as pockets.

Does this apply to professions spawning with LBV/BVs as well?

**Comment by bombasticSlacks** [comment]
It can! I'll add it to the scope

**Comment by bombasticSlacks** [comment]
Your going to have to be more specific on what you are referring to. If it is the cheap security vest of the same caliber as an airsoft vest, no, under no circumstances is cheap garbage armor going to be better than any form of NIJ 3+ armor.

If you want to find me a reference to a real stab vest that is better than NIJ 3a then I can add it.

**Comment by Shodan14** [comment]
> Your going to have to be more specific on what you are referring to. If it is the cheap security vest of the same caliber as an airsoft vest, no, under no circumstances is cheap garbage armor going to be better than any form of NIJ 3+ armor.
> 
> If you want to find me a reference to a real stab vest that is better than NIJ 3a then I can add it.
I don't have enough knowledge to go into specific models. My general understanding is that they reason we have stab vests (particularly in UK, NZ) is that regular ballistic vests aren't very effective against slow moving attacks like knife stabs, might be [better explained here](http://www.military-today.com/life/stab_vest_vs_bullet_proof_vest_the_definitive_guide.htm).

I'm not entirely sure how good an "airsoft vest" is at anything, but what's the rationale for the high cut protection on the ballistic vests?

**Comment by bombasticSlacks** [comment]
Kevlar has great slash resistance. That is why it's used for cut proof clothing, gloves and chaps for chainsaws.

Fundamentally though I didn't set the cut resistance, whoever implemented Kevlar did. I'm just describing the materials each thing is made of correctly. If you think any materials info is wrong I would encourage you to change it in a separate PR.

**Comment by Shodan14** [comment]
> Fundamentally though I didn't set the cut resistance, whoever implemented Kevlar did. I'm just describing the materials each thing is made of correctly. If you think any materials info is wrong I would encourage you to change it in a separate PR.

Yup, I agree, I think that's probably where the actual problem lies. You can use kevlar for both kinds of vests, but there's a reason they aren't interchangeable.

File: data/json/itemgroups/custom_lbvs.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      { "item": "gadget_pouch", "prob": 10 },
+      { "item": "h2o_pouch", "prob": 10 },
+      { "item": "tool_pouch", "prob": 10 },
+      { "item": "flashlight_pouch", "prob": 20 },
+      { "item": "tactical_holster", "prob": 20 },
+      { "item": "gas_mask_pouch", "prob": 10 },
+      { "item": "tactical_grenade_pouch", "prob": 40 },
+      { "item": "grenade_pouch", "prob": 10 },
+      { "item": "triple_stacker_pouch", "prob": 10 },
+      { "item": "dump_pouch", "prob": 20 }
```
Comment (bombasticSlacks) [inline data/json/itemgroups/custom_lbvs.json:28]:
```suggestion
      { "item": "dump_pouch", "prob": 20 },
      { "item": "canteen_pouch", "prob": 30 }
```

**Comment by Ilysen** [comment]
The SWAT Officer profession appears to have no pants after this PR; the ballistic vest also conflicts with the tac vest that they've had in the past. I'm unsure about other professions.

**Comment by bombasticSlacks** [comment]
LOL of course because they never had pants just armor.  I can clean up the tac vest and give the poor people some pants in a bit.

**Pull request merged**